### PR TITLE
Get r squared metric

### PR DESCRIPTION
**Pull request by LauraZM9**
[[Epic2] C - Run and store accuracy metrics for 'Estimate Jobs' Residential role calculation model](https://trello.com/c/stA2iFiX)
# Description

- Get R<sup>2</sup> metric for care homes with historical model
- Collect data from when the metric was generated:
  - `percentage_data` - percentage of data used in the model
  - `latest_snapshot` - latest snapshot date in the dataset
  - `job_id` - job id from Glue
  - `generated_metric_date` - timestamp when the metric was generated
  - `model_version` - version of the model used
- Write data in S3 

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket

### CODE REVIEW

File: jobs/estimate_job_counts.py [hunk 1]
```diff
@@ -243,11 +261,53 @@ def model_care_home_with_historical(locations_df, features_df, model_path):
...
+    print("Calculating R Squared (R2) = %g" % r2)
+
+    return r2
+
+
+def write_metrics_df(
+    metrics_destination, r2, data_percentage, model_version, latest_snapshot, job_id
+):
+    spark = utils.get_spark()
+    schema = StructType(
```
Comment (elcorbs) [inline jobs/estimate_job_counts.py:291]: Can we have the name of the model as well, I assume we'll store the metrics for all models together

File: jobs/estimate_job_counts.py [hunk 2]
```diff
@@ -33,6 +40,9 @@ def main(
     prepared_locations_features,
     destination,
     care_home_model_directory,
+    metrics_destination,
+    JOB_RUN_ID,
+    JOB_NAME,
```
Comment (elcorbs) [inline jobs/estimate_job_counts.py:45]: These don't need to be capitalised

File: jobs/estimate_job_counts.py [hunk 3]
```diff
@@ -68,11 +78,21 @@ def main(
...
         f"{care_home_model_directory}{latest_model_version}/",
     )
+    latest_snapshot = utils.get_max_snapshot_date(locations_df)
+    write_metrics_df(
+        metrics_destination,
+        r2=metrics_info["r2"],
+        data_percentage=metrics_info["data_percentage"],
+        model_version=latest_model_version,
+        latest_snapshot=latest_snapshot,
+        job_id=JOB_RUN_ID,
```
Comment (elcorbs) [inline jobs/estimate_job_counts.py:93]: This parameter should be called job run id as well

File: jobs/estimate_job_counts.py [hunk 4]
```diff
@@ -253,11 +273,60 @@ def model_care_home_with_historical(locations_df, features_df, model_path):
...
+    row = [(r2, data_percentage, latest_snapshot, job_id, job_name, model_version)]
+    df = spark.createDataFrame(row, schema)
+    df = df.withColumn("generated_metric_date", F.current_timestamp())
+
+    print(f"Writing model metrics as parquet to {metrics_destination}")
+    utils.write_to_parquet(
+        df,
+        metrics_destination,
+        append=True,
+        partitionKeys=["model_version"],
```
Comment (elcorbs) [inline jobs/estimate_job_counts.py:328]: I think we should partition by model name as well as model version

**Review by JimmySunDance — APPROVED** [review]
Happy with this

**Pull request merged**